Source/FieldSolver/ElectrostaticSolver_ME.cpp

```diff
@@ -0,0 +1,658 @@
+        WarpXParticleContainer& species = mypc->GetParticleContainer(ispecies);
+        bool const local = true;
+        bool const reset = false;
+        bool const do_rz_volume_scaling = false;
+        species.DepositCharge(rho_fp, local, reset, do_rz_volume_scaling);
+    }
+    for (int lev = 0; lev <= max_level; lev++) {
+        ApplyFilterandSumBoundaryRho (lev, lev, *rho_fp[lev], 0, 1);
+    }
+#ifdef WARPX_DIM_RZ
+    for (int lev = 0; lev <= max_level; lev++) {
+        ApplyInverseVolumeScalingToChargeDensity(rho_fp[lev].get(), lev);
+    }
+#endif
+}
+
+void
+WarpX::AddSpaceChargeFieldLabFrame ()
+{
+
+#ifdef WARPX_DIM_RZ
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(n_rz_azimuthal_modes == 1,
+                                     "Error: RZ electrostatic only implemented for a single mode");
+#endif
+
+    // beta is zero in lab frame
+    // Todo: use simpler finite difference form with beta=0
+    std::array<Real, 3> beta = {0._rt};
+
+    // Compute the potential phi, by solving the Poisson equation
+    computePhi( rho_fp, phi_fp, beta, self_fields_required_precision, self_fields_max_iters );
+
+}
+
+void
+WarpX::ComputeSpaceChargeField (bool const reset_fields)
+{
+    if (reset_fields) {
+        // Reset all E and B fields to 0, before calculating space-charge fields
+        for (int lev = 0; lev <= max_level; lev++) {
+            for (int comp=0; comp<3; comp++) {
+                Efield_fp[lev][comp]->setVal(0);
+                Bfield_fp[lev][comp]->setVal(0);
+            }
+        }
+    }
+
+    // beta is zero in lab frame
+    // Todo: use simpler finite difference form with beta=0
+    std::array<Real, 3> beta = {0._rt};
+
+    // Compute the corresponding electric and magnetic field, from the potential phi
+    computeE( Efield_fp, phi_fp, beta );
+    computeB( Bfield_fp, phi_fp, beta );
+
+    // Transfer fields from 'fp' array to 'aux' array.
+    // This is needed when using momentum conservation
+    // since they are different arrays in that case.
+    UpdateAuxilaryData();
+    FillBoundaryAux(guard_cells.ng_UpdateAux);
+}
+
+/* Compute the potential `phi` by solving the Poisson equation with `rho` as
+   a source, assuming that the source moves at a constant speed \f$\vec{\beta}\f$.
+   This uses the amrex solver.
+   \param[in] rho The charge density a given species
+   \param[out] phi The potential to be computed by this function
+   \param[in] beta Represents the velocity of the source of `phi`
+*/
+void
+WarpX::computePhi (const amrex::Vector<std::unique_ptr<amrex::MultiFab> >& rho,
+                   amrex::Vector<std::unique_ptr<amrex::MultiFab> >& phi,
+                   std::array<Real, 3> const beta,
+                   Real const required_precision,
+                   int const max_iters) const
+{
+#ifdef WARPX_DIM_RZ
+    computePhiRZ( rho, phi, beta, required_precision, max_iters );
+#else
+    computePhiCartesian( rho, phi, beta, required_precision, max_iters );
+#endif
```

Comment:
Please also incorporate @MKieburtz's changes here that allows us to set the field solve verbosity.